src/settings/components/TermsConfig/TermsConfigForm.js

```diff
@@ -29,14 +32,29 @@ class TermsConfigForm extends React.Component {
     });
   }
 
+  sendCalloutTermInUse = () => {
+    this.callout.sendCallout({
+      type: 'error',
+      message: <SafeHTMLMessage id="ui-licenses.settings.terms.callout.delete.termInUse" values={{ className: css.error }} />,
+      timeout: 0,
+    });
+  }
+
   handleDelete = (...rest) => {
     return this.props.onDelete(...rest)
       .then(() => this.sendCallout('delete', 'success'))
       .catch(response => {
         // Attempt to show an error message if we got JSON back with a message.
         // If json()ification fails, show the generic error callout.
         response.json()
-          .then(error => this.sendCallout('delete', 'error', error.message))
+          .then((error) => {
+            const pattern = new RegExp('violates foreign key constraint.*is still referenced from table', 'ms');
```

Comment:
I am not a 100% sure, but I dont think we need the `m` flag here as that flag only affects the behavior of `^` and `$`